src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java

```diff
@@ -68,17 +68,30 @@ public LocalUpdateSyncRequest(String thingName,
      * @throws IOException if unable to serialize the update document payload bytes.
      */
     public void merge(LocalUpdateSyncRequest other) throws IOException {
-        Optional<JsonNode> oldValueJson = JsonUtil.getPayloadJson(updateDocument);
-        Optional<JsonNode> newValueJson = JsonUtil.getPayloadJson(other.getUpdateDocument());
-        if (!oldValueJson.isPresent() && newValueJson.isPresent()) {
+        Optional<JsonNode> currDocJson = JsonUtil.getPayloadJson(updateDocument);
+        Optional<JsonNode> updateDocJson = JsonUtil.getPayloadJson(other.getUpdateDocument());
+
+        if (!currDocJson.isPresent() && updateDocJson.isPresent()) {
             updateDocument = other.getUpdateDocument();
             return;
         }
-        if (!newValueJson.isPresent()) {
+        if (!updateDocJson.isPresent()) {
             return;
         }
-        JsonMerger.merge(oldValueJson.get(), newValueJson.get());
-        updateDocument = JsonUtil.getPayloadBytes(oldValueJson.get());
+
+        JsonNode oldValueJson = currDocJson.get();
+        JsonNode newValueJson = updateDocJson.get();
+
+        if (JsonUtil.hasVersion(oldValueJson) && JsonUtil.hasVersion(newValueJson)
+                && JsonUtil.getVersion(oldValueJson) > JsonUtil.getVersion(newValueJson)) {
+            // updates received out of order,
```

Comment:
can you add a debug log here saying it received version X after receiving version X+1

Reply:
done